src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component;
+
+/**
+ * Interface Factory
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Factory
+{
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *      An Interruptive Modal disrupts the user in critical situation, forcing him or her to focus on the task at hand.
+     *   composition: >
+     *      The Modal states why this situation needs attention and may point out consequences.
+     *   effect: >
+     *      All controls of the original context are inaccessible until the Modal is completed. Upon completion the user returns to the original context.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *        Due to the heavily disruptive nature of this type of Modal it MUST be restricted to critical situations (e.g. loss of data).
+     *     2: >
+     *        All delete-actions are critical situations and MUST be implemented as an Interruptive Modal.
```

Comment:
As soon as you delete it on the persistent layer (not by just changing the UI on the frontend), I would say so. However, we could make that a SHOULD, or give us a better rule proposal.